lib/winrm/http/transport.rb

```diff
@@ -47,6 +49,26 @@ def send_request(message)
         resp = @httpcli.post(@endpoint, message, hdr)
```

Comment:
At the time of this call to POST our first message, we don't know if the fingerprint will match, but we go ahead and send the http request.

We have already connected once in ssl_peer_fingerprint_verification! but if we had a complex malicious-in-the-middle scenario, they could wait for the initial verify then redirect all subsequent connections.

I'm not sure of all the initial http soap requests winrm makes, but I'd like to verify that we could fail after that first request and still be safe. It's obviously fine for negotiate and kerberos, but I'm unsure if the first request for plaintext would send anything.